### PR TITLE
Task ys/j194 user with role admin can delete podcast episode functionality

### PR DESCRIPTION
**Pull request by yaroslav207**

### CODE REVIEW

**Review by what1s1ove — CHANGES_REQUESTED** [review]
we also need to show delete button for admins on the UI

File: backend/src/middlewares/check-user-owner/check-user-episode-owner.middleware.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import { RequestHandler } from 'express';
 import { HttpError } from '~/exceptions/exceptions';
-import { ErrorMessage, HttpCode } from '~/common/enums/enums';
+import { ErrorMessage, HttpCode, UserRole } from '~/common/enums/enums';
 import { episode as episodeService } from '~/services/services';
 
 const checkUserEpisodeOwner = (): RequestHandler => {
   const handler: RequestHandler = async (req, _res, next) => {
+    if (req.user?.role === UserRole.MASTER) {
+      next();
```
Comment (what1s1ove) [inline backend/src/middlewares/check-user-owner/check-user-episode-owner.middleware.ts]:
```suggestion
      return next();
```

File: backend/src/middlewares/check-user-owner/check-user-podcast-owner.middleware.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
 import { RequestHandler } from 'express';
 import { HttpError } from '~/exceptions/exceptions';
-import { ErrorMessage, HttpCode } from '~/common/enums/enums';
+import { ErrorMessage, HttpCode, UserRole } from '~/common/enums/enums';
 import { podcast as podcastService } from '~/services/services';
 
 const checkUserPodcastOwner = (): RequestHandler => {
   const handler: RequestHandler = async (req, _res, next) => {
+    if (req.user?.role === UserRole.MASTER) {
+      next();
```
Comment (what1s1ove) [inline backend/src/middlewares/check-user-owner/check-user-podcast-owner.middleware.ts]:
```suggestion
      return next();
```

File: frontend/src/components/episode/episode.tsx [hunk 3]
```diff
@@ -55,6 +55,7 @@ const Episode: React.FC = () => {
   const hasRecord = Boolean(episode?.record);
   const isStaging = episode?.status === EpisodeStatus.STAGING;
   const isOwner = user?.id === episode?.userId;
+  const isMaster = user?.role === UserRole.MASTER;
   const isPlayerLoaded = playerStatus === DataStatus.FULFILLED;
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx:61]:
```suggestion
const isAllowDelete = isOwner || isMaster;

```

File: frontend/src/components/episode/episode.tsx [hunk 4]
```diff
@@ -122,13 +123,15 @@ const Episode: React.FC = () => {
...
               <div className={styles.descriptionWrapper}>
                 {isOwner && (
+                  <Link
+                    to={`${AppRoute.PODCASTS}/${episode.podcastId}${AppRoute.EPISODES_EDIT}/${episode.id}`}
+                    className={styles.editLink}
+                  >
+                    <span className="visually-hidden">Edit episode</span>
+                  </Link>
+                )}
+                {(isOwner || isMaster) && (
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
                {isAllowDelete && (
```

File: frontend/src/components/podcast/podcast.tsx [hunk 5]
```diff
@@ -68,11 +70,13 @@ const Podcast: React.FC = () => {
             )}
             <div className={styles.podcastInfoWrapper}>
               {isOwner && (
+                <Link
+                  to={`${AppRoute.PODCASTS_EDIT}/${podcast.id}`}
+                  className={styles.editLink}
+                />
+              )}
+              {(isOwner || isMaster) && (
```
Comment (what1s1ove) [inline frontend/src/components/podcast/podcast.tsx]:
```suggestion
              {isAllowDelete && (
```
admin can **only** delete

File: frontend/src/components/episode/episode.tsx (hunk 4 above)
Comment (what1s1ove) [reply frontend/src/components/episode/episode.tsx]: admin can **only** delete

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by JuliaDzyuba — APPROVED** [review]

**Review by EugeniusGR — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**